libs/generic-view/feature/src/lib/modals/data-migration-error-modal.tsx

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+import React, { FunctionComponent, useEffect } from "react"
+import { Modal, TransferErrorModal } from "generic-view/ui"
+import { useDispatch, useSelector } from "react-redux"
+import {
+  DataMigrationPercentageProgress,
+  selectActiveApiDeviceId,
+  selectDataMigrationStatus,
+  setDataMigrationFeatures,
+  setDataMigrationProgress,
+  setDataMigrationPureDbIndexing,
+  setDataMigrationSourceDevice,
+  setDataMigrationStatus,
+  setDeviceErrorModalOpened,
+} from "generic-view/store"
+import { ModalLayers } from "Core/modals-manager/constants/modal-layers.enum"
+
+export const DataMigrationErrorModal: FunctionComponent = () => {
+  const dispatch = useDispatch()
+  const activeDevice = useSelector(selectActiveApiDeviceId)
+  const dataMigrationStatus = useSelector(selectDataMigrationStatus)
+  const opened = dataMigrationStatus === "FAILED" && !activeDevice
+
+  const onClose = () => {
+    dispatch(setDataMigrationStatus("IDLE"))
+    dispatch(setDataMigrationProgress(DataMigrationPercentageProgress.None))
+    dispatch(setDataMigrationFeatures([]))
+    dispatch(setDataMigrationSourceDevice(undefined))
+    dispatch(setDataMigrationPureDbIndexing(false))
+    dispatch(setDeviceErrorModalOpened(false))
```

Comment:
⚪ (question)
Is this the only place where the state is reset? Maybe we should consolidate this group of actions into one place?

Reply:
There are more places like that but depending on context, different actions are dispatched so I actually switched from single thunk to multiple dispatches because of it ;)